### PR TITLE
8 - Make more use of `let_chains`

### PR DESCRIPTION
**Pull request by c410-f3r**
Continuation of #94376.

cc #53667

### CODE REVIEW

File: compiler/rustc_infer/src/infer/error_reporting/nice_region_error/trait_impl_difference.rs [hunk 1]
```diff
@@ -23,51 +23,45 @@ impl<'a, 'tcx> NiceRegionError<'a, 'tcx> {
...
-            _,
-        ) = error.clone()
+                _,
+                var_origin,
+                sub_origin,
+                _sub,
+                sup_origin,
+                _sup,
+                _,
+            ) = error.clone()
```
Comment (jackh726) [inline compiler/rustc_infer/src/infer/error_reporting/nice_region_error/trait_impl_difference.rs:33]: Is this indentation too far in? I wouldn't have expected this to change.

**Review by jackh726 — APPROVED** [review]
Just one nit, then r=me

I will say, I don't think I've seen a single one of these PRs where replacing with `let_chains` *isn't* a very clear improvement.

File: compiler/rustc_infer/src/infer/error_reporting/nice_region_error/trait_impl_difference.rs (hunk 1 above)
Comment (c410-f3r) [reply compiler/rustc_infer/src/infer/error_reporting/nice_region_error/trait_impl_difference.rs]: Folded these in a single line

**Pull request merged**